### PR TITLE
Don't show upsell if a term variant of the upsold product is already owned

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

Let's say you own a product of a specific term (e.g. Backup Daily, monthly). When purchasing another product that upsells it (e.g. Scan), with a different term selected, the upsell is still shown. In this PR, we make sure to not show the upsell in that case.

Note: upselling a product with a superior term is maybe something we'd like to do in the future, but the UI doesn't handle this case well for now.

### Testing instructions

- Select a self-hosted Jetpack site that has no product or plan
- Visit the _Plans_ section with the offer flow enabled
- Try to purchase Scan, with any term
- Verify that you see the upsell for Backup Daily
- Select a site that has Backup Daily, with a monthly term, and go back to the _Plans_ page
- Try to purchase Scan, with a yearly term
- Verify that no upsell is shown
- Do the same with Backup Daily on a yearly term, and Scan on a monthly term

### CODE REVIEW

**Review by rcanepa — CHANGES_REQUESTED** [review]
For some unknown reason, it didn't work when I tested it having Jetpack Scan yearly on my site. I could see the Jetpack Scan upsell when I followed the Jetpack Backup Daily monthly path. This is what I saw:

Kapture 2020-09-10 at 17 11 16

**Comment by rcanepa** [comment]
The approach is great. I think we may be able to re-use what you did here to accomplish 1169247016322522-as-1191532836382393. Basically, that ticket calls for having a way to switch between terms of the same product, e.g., `jetpack_scan` to `jetpack_scan_monthly` and vice versa.

**Comment by monsieur-z** [comment]
Up for review again.

File: client/my-sites/plans-v2/use-has-product-upsell.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import type { ProductSlug } from 'lib/products-values/types';
+
+/**
+ * Returns a function that checks if a product upsells another product. Note that a product that
+ * should be upsold won't be upsold is the current user owns it, owns a similar product with a
+ * different term, or owns a product that overrides it.
+ *
+ * @returns {Function} Callback
+ */
+export default function useHasProductUpsell(): ( arg0: ProductSlug ) => boolean {
```
Comment (rcanepa) [inline client/my-sites/plans-v2/use-has-product-upsell.ts:27]: This is pretty cool.

**Review by rcanepa — APPROVED** [review]
LGTM now!

**Pull request merged**